lib/salus/scanners/osv/maven_osv.rb

```diff
@@ -0,0 +1,150 @@
+    DEFAULT_SOURCE = "https://osv.dev/list".freeze
+    DEFAULT_SEVERITY = "MODERATE".freeze
+    GITHUB_DATABASE_STRING = "Github Advisory Database".freeze
+    MAVEN_OSV_ADVISORY_URL = "https://osv-vulnerabilities.storage.googleapis.com/"\
+        "Maven/all.zip".freeze
+
+    def should_run?
+      @repository.pom_xml_present?
+    end
+
+    def run
+      dependencies = find_dependencies
+      if dependencies.empty?
+        err_msg = "MavenOSV: Failed to parse any dependencies from the project."
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      @osv_vulnerabilities ||= fetch_vulnerabilities(MAVEN_OSV_ADVISORY_URL)
+      if @osv_vulnerabilities.nil?
+        err_msg = "No vulnerabilities found to compare."
+        bugsnag_notify("MavenOSV: #{err_msg}")
+        return report_error("MavenOSV: #{err_msg}")
+      end
+
+      # Fetch vulnerable dependencies.
+      # Dedupe and select Github Advisory over other sources if available.
+      results = []
+      grouped = match_vulnerable_dependencies(dependencies).group_by { |d| d[:ID] }
+      grouped.each do |_key, values|
+        vuln = {}
+        values.each do |v|
+          vuln = v if v[:Database] == GITHUB_DATABASE_STRING
+        end
+        results.append(vuln.empty? ? values[0] : vuln)
+      end
+
+      # Report scanner status
+      return report_success if results.empty?
+
+      report_failure
+      log(JSON.pretty_generate(results))
+    end
+
+    # Match if dependency version found is in the range of
+    # vulnerable dependency found.
+    def version_matching(version, version_ranges)
+      vulnerable_flag = false
+      version_found = SemVersion.new(version)
+      # If version range length is 1, then no fix available.
+      if version_ranges.length == 1
+        introduced = SemVersion.new(
+          version_ranges[0]["introduced"]
+        )
+        vulnerable_flag = true if version_found >= introduced
+      # If version range length is 2, then both introduced and fixed are available.
+      elsif version_ranges.length == 2
+        introduced = SemVersion.new(
+          version_ranges[0]["introduced"]
+        )
+        fixed = SemVersion.new(
+          version_ranges[1]["fixed"]
+        )
+        vulnerable_flag = true if version_found >= introduced && version_found < fixed
+      end
+
+      vulnerable_flag
+    end
+
+    # Compare vulnerabilities found with dependencies found
+    # and return vulnerable dependencies
+    def match_vulnerable_dependencies(dependencies)
+      results = []
+      dependencies.each do |dependency|
+        lib = "#{dependency['group_id']}:#{dependency['artifact_id']}"
+
+        unless dependency['version'].nil?
+          version = dependency['version']
+          # If version is of the format '${deps.version}', log it.
+          if version.match(/\${(.*)\.version}/)
+            bugsnag_notify("MavenOSV: Found #{lib}:#{version} with incompatible format.")
```

Comment:
Does `${deps.version}` indicate the pom is mis-configured?

Reply:
Not misconfigured but we aren't able to match a version for it. With the properties parsing introduced I don't think it should be triggered but still adding in just in case if we encounter it.

Reply:
Ok. Hopefully we don't suddenly get a project with tons of `${deps.version}` and then get tons of bugsnag pings.

But should you instead check for anything that start with `$`, they could put other variable names inside, right?

Reply:
I wonder if there is a tool that gives the actual version of `${deps.version}`. Though may be outside the scope of this PR.

Reply:
There might be but I am not too worried about it since they won't show up under `<dependencies>` tag. So the main thing I saw which we needed to handle was `${deps.version}` which gets addressed while parsing `pom.xml ` and added a check in case if we encounter any issues.